test/teiserver_web/tachyon/matchmaking_test.exs

```diff
@@ -110,13 +225,73 @@ defmodule Teiserver.Tachyon.MatchmakingTest do
           id: id,
           name: id,
           team_size: 0,
-          team_count: 2
+          team_count: 2,
+          engines: engine_versions(),
+          games: game_versions(),
+          maps: map_names(id)
         })
         |> Teiserver.Matchmaking.QueueServer.start_link()
 
       assert %{"status" => "failed", "reason" => "invalid_request"} =
                Tachyon.join_queues!(client, [id])
     end
+
+    test "empty engines", %{client: client} do
+      id = "emptyengines"
+
+      {:ok, _} =
+        Teiserver.Matchmaking.QueueServer.init_state(%{
+          id: id,
+          name: id,
+          team_size: 1,
+          team_count: 2,
+          engines: [],
+          games: game_versions(),
+          maps: map_names(id)
+        })
+        |> Teiserver.Matchmaking.QueueServer.start_link()
```

Comment:
Why not use `QueueSupervisor.start_queue!(state)`? This way you guarantee that this setup is faithful to the real thing.

Reply:
Copied from the "too many players" test above, will change.

Reply:
Changed.